core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandlerBase.java

```diff
@@ -755,13 +764,24 @@ public void cancel() {
       }
     }
 
-    private void trackNodeError(Node node, Throwable error) {
+    private void trackNodeError(Throwable error) {
       long latencyNanos = System.nanoTime() - this.start;
       context
           .getRequestTracker()
           .onNodeError(statement, error, latencyNanos, executionProfile, node);
     }
 
+    private void maybeUpdateSpeculativeExecutionsInFlightCounter(int amount) {
+      if (this.execution > 0) {
```

Comment:
why we need that conditional logic here? How can be a situation when we don't have execution > 0 but the execution finished executing: `maybeUpdateSpeculativeExecutionsInFlightCounter`?

Reply:
If `execution == 0` then this is the first (i.e., non-speculative execution of the request); for `execution > 0`, we are sure that this is a speculative execution.